### ISSUE TITLE
Minor issues

### ISSUE THREAD

**Issue by woo1127** [AutumnPizazz/the-Coexistence-in-harmony#17]
Hi Author, I must say this is one of the best Unciv mod I ever see. I urge you to continue supporting its development in the future, it bring a lot fun to me 😁. The mod's features align more closely with the original civ6 compare to others civ6 mod of Unciv lol.

#### Issues that I currently faced during the game:
- Download troubles: I encountered difficulties downloading the mod in the game. I currently resolved it by manually copy paste this repo link and adding the suffix `/tree/v23.10.30`, for example `https://github.com/AutumnPizazz/CoeHarMod/tree/v23.10.30`.
- Unrepairable pillaged improvement. This issue become so frustrating when barbarian always pillage the improvement.
- Worker types consolidation. It would be great to consolidate workers into a single type instead of having distinct categories like Farmer and Miner. I know this is impossible for now due to the current limitation of supported unique parameters, perhaps a temporary solution could be reusing the worker concept from Civ 5?

This mod could become more popular if it expands beyond BiliBili, maybe could share it to unciv reddit and discord too ?

**Comment by AutumnPizazz** [comment]
> * Download troubles: I encountered difficulties downloading the mod in the game. I currently resolved it by manually copy paste this repo link and adding the suffix `/tree/v23.10.30`, for example `https://github.com/AutumnPizazz/CoeHarMod/tree/v23.10.30`.

When you update the Unciv game program to version 4.8.18 or later, you should be able to resolve this issue (as I mentioned in [this issue](https://github.com/yairm210/Unciv/issues/10447)).



> * Unrepairable pillaged improvement. This issue become so frustrating when barbarian always pillage the improvement.

Haha! That's one of the differences between CoeHarMod and Civ6. In CoeHarMod, pillaged improvements can only be repaired by Traders. (Here, Traders can only build roads and repair pillaged improvements.)

> * Worker types consolidation. It would be great to consolidate workers into a single type instead of having distinct categories like Farmer and Miner. I know this is impossible for now due to the current limitation of supported unique parameters, perhaps a temporary solution could be reusing the worker concept from Civ 5?

I don't want merging Workers to be the outcome, at least not until Uniques statements support it like in Civ6. Because I really hated the mechanic in Civ5 where Workers took up a lot of turns to build improvements. I think the current situation is good.

**Comment by AutumnPizazz** [comment]
> This mod could become more popular if it expands beyond BiliBili, maybe could share it to unciv reddit and discord too ?

Haha! The CoeHarMod player community is not limited to Bilibili. On the contrary, it's just a very small platform, to the extent that I often overlook it.
The CoeHarMod player community is mostly on Tencent QQ groups. We have nearly two hundred members there, with nearly a thousand messages exchanged every day, and there are almost daily multiplayer matches. It's a pleasure to see everyone using my mod for multiplayer.

**Comment by AutumnPizazz** [comment]
And I created [a section for CoeHarMod](https://discord.com/channels/586194543280390151/1172561627967660163) on Discord a week ago.

**Comment by woo1127** [comment]
Thank for answering my question, as what you told I upgrade to latest version and my problem solved 😁

<br>

> I don't want merging Workers to be the outcome, at least not until Uniques statements support it like in Civ6. Because I really hated the mechanic in Civ5 where Workers took up a lot of turns to build improvements. I think the current situation is good.

Ok I respect your decision.

<br>

> The CoeHarMod player community is mostly on Tencent QQ groups. We have nearly two hundred members there, with nearly a thousand messages exchanged every day, and there are almost daily multiplayer matches. It's a pleasure to see everyone using my mod for multiplayer.

I really want to join your QQ group, but after downloaded the app I encountered the issue as shown on the pic when attempting to sign up. I try various of my phone numbers, I also try using the virtual Chinese phone number but still not working, I search the web and found out that this is a way QQ used to restrict access for overseas users.

![qq error](https://github.com/AutumnPizazz/CoeHarMod/assets/71725118/c777bed2-df78-4ed6-a4ca-0144764052e9)

**Comment by woo1127** [comment]
https://www.zhihu.com/question/386018227

**Comment by AutumnPizazz** [comment]
I'm very sad that QQ has restricted the use for foreigners, but I don't have any good solutions to solve this problem. QAQ

**Comment by AutumnPizazz** [comment]
https://ssl.zc.qq.com/v3/index-en.html?type=0
Is it useful?

**Comment by woo1127** [comment]
nope, still the same issue 😥
but we foreigner can use wechat, do you want to move your group to wechat ? 😂

**Comment by AutumnPizazz** [comment]
Migrating a group chat of 200 people to WeChat is an extremely troublesome task.

**Comment by AutumnPizazz** [comment]
There are many QQ accounts without bound phone numbers being sold on Taobao. I believe this is one possible solution, but it also comes with the risk of being scammed.

**Comment by AutumnPizazz** [comment]
https://tb.alicdn.com/

**Comment by AutumnPizazz** [comment]
Due to sensitive word filtering, you should search for "球球号" instead of "QQ号".

**Comment by woo1127** [comment]
Thanks for the solution, my mom always shop at taobao lol

**Comment by AutumnPizazz** [comment]
"QQ premium accounts"（QQ靓号） are also a solution. However, after checking the prices, each premium account requires an expenditure of approximately 120 RMB, which is too expensive (at least for me).

**Comment by AutumnPizazz** [comment]
<img width="1280" alt="image" src="https://github.com/AutumnPizazz/CoeHarMod/assets/114050365/c9c3caa3-a199-4c46-9955-d7c7db7a968d">

**Comment by AutumnPizazz** [comment]
> There are many QQ accounts without bound phone numbers being sold on Taobao. I believe this is one possible solution, but it also comes with the risk of being scammed.

The QQ accounts obtained through this method often do not support login from overseas, which is very frustrating. (After consulting with the customer service)

**Comment by AutumnPizazz** [comment]
Now the much more worker problems have been solved happily!

**Comment by woo1127** [comment]
> Now the much more worker problems have been solved happily!

Yes I have saw your new announcement on Discord today, worker can now build almost all fundamental improvements, I am impressed with the way you solved the problem. Thanks so much for taking my suggestion so seriously.

At the moment I don't plan on purchasing any QQ account yet, aside from the risk factor, one of the reason I play Unciv is its free accessibility, and I want prevent any unnecessary expenditure.  I hope that someday QQ can open to oversea this big market.

I am now recommend and invite my fellow multiplayers on Discord to play this mod together, I hope to attract more players on Discord to participate this in the future. One hindrance I found for players to interesting in this mod lies in its name, CoeHarMod, They have a false assumption that this mod is only solely with Chinese history, ignoring the fact that this mod has rich Civ 6 contents too.
